connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/LogReporter.java

```diff
@@ -50,17 +53,18 @@ public LogReporter(ConnectorTaskId id, ConnectorConfig connConfig, ErrorHandling
      * @param context the processing context.
      */
     @Override
-    public void report(ProcessingContext context) {
+    public Future<RecordMetadata> report(ProcessingContext context) {
         if (!connConfig.enableErrorLog()) {
-            return;
+            return CompletableFuture.completedFuture(null);
         }
 
         if (!context.failed()) {
-            return;
+            return CompletableFuture.completedFuture(null);
         }
 
         log.error(message(context), context.error());
         errorHandlingMetrics.recordErrorLogged();
+        return CompletableFuture.completedFuture(null);
```

Comment:
What about creating:
```
    private static final Future<RecordMetadata> COMPLETED = CompletableFuture.completedFuture(null);
```

and then returning that instance in all of these places. Since it's already completed, immutable, and we don't allow cancellation, it should be fine to reuse in this `LogReporter`.